src/t_list.c

```diff
@@ -790,13 +790,12 @@ void lremCommand(client *c) {
     listTypeReleaseIterator(li);
 
     if (removed) {
-        signalModifiedKey(c,c->db,c->argv[1]);
         notifyKeyspaceEvent(NOTIFY_LIST,"lrem",c->argv[1],c->db->id);
-    }
-
-    if (listTypeLength(subject) == 0) {
-        dbDelete(c->db,c->argv[1]);
-        notifyKeyspaceEvent(NOTIFY_GENERIC,"del",c->argv[1],c->db->id);
+        if (listTypeLength(subject) == 0) {
+            dbDelete(c->db,c->argv[1]);
+            notifyKeyspaceEvent(NOTIFY_GENERIC,"del",c->argv[1],c->db->id);
+        }
+        signalModifiedKey(c,c->db,c->argv[1]);
```

Comment:
Here your modification change program logic. Before you change, dbDelete() could be called when removed  variable = 0 and listTypeLength(subject) == 0, and after you change it, if removed = 0, then dbDelete() will not be called. Coud you please check if this situation could happen?   Thanks

Reply:
we don't expect empty keys, so it can't be that no element was removed, but the list is now empty.
LGTM